app/models/atmosphere/action.rb

```diff
@@ -0,0 +1,24 @@
+module Atmosphere
+  class Action < ActiveRecord::Base
+
+    belongs_to :appliance,
+               class_name: 'Atmosphere::Appliance'
+
+    has_many :action_logs,
+             dependent: :destroy,
+             class_name: 'Atmosphere::ActionLog'
+
+    def log(message, level = :info)
```

Comment:
Yeap, message should be added here. What is more you can use more user friendly syntax here:

```ruby
action_logs.create(message: message, log_level: level)
```

It will automatically set correct relation between `action` and `action_log`